uberpoet/cpprojectgen.py

```diff
@@ -107,37 +108,18 @@ def gen_app(self, app_node, node_list, target_swift_loc, target_objc_loc, loc_js
                 total_code_units += l.code_units
 
             total_loc = target_swift_loc + target_objc_loc
-
             swift_module_count_percentage = round(float(target_swift_loc) / total_loc, 2)
-            objc_module_count_percentage = round(float(target_objc_loc) / total_loc, 2)
-
-            # Splits the generated module list across the percentage of LOC that needs to be allocated between ObjC
```

Comment:
so...I updated this logic since I want to loop through the entire node list in order to keep track of the nodes we've created sources for and store that in the `module_index`.

This actually simplified this logic imo, basically we loop through the whole list now instead of splitting it by percentage and then depending on how far off we are (based on percentages) we switch the language to generate for.

Semantically this is the same as before.

Reply:
I'd love to maybe update this instead of just going in sequence to randomize up to the maximum percentage.

So instead:
```
MockLib0 -> Swift
MockLib1 -> Swift
MockLib2 -> Swift
MockLib3 -> ObjC
MockLib4 -> ObjC
MockLib5 -> ObjC
```

We can do maybe interleaved:
```
MockLib0 -> Swift
MockLib1 -> ObjC
MockLib2 -> Swift
MockLib3 -> ObjC
MockLib4 -> Swift
MockLib5 -> ObjC
```

But this can be done in a follow up PR.

Reply:
I will update the Blaze project generator to follow this logic once I get more feedback in general for this PR.

Reply:
You're testing 2 different things with interleaved or not interleaved (more or less objc<->swift interop).  It should almost be an option.   I think whatever you like to do is fine by me although, I have no strong opinion on it, just some ideas.

Also another thing we could add to dot files is determining if they are objc or swift modules, that behavior in real life apps is pretty valuable to capture for simulations.

Good job on the simplification.

Reply:
> Also another thing we could add to dot files is determining if they are objc or swift modules, that behavior in real life apps is pretty valuable to capture for simulations.

ah yes we can add that in the DOT representation perhaps by an attribute within each node.

For now, you can successfully do this via the loc file path which accepts a language and will be parsed by Uber Poet.

> You're testing 2 different things with interleaved or not interleaved (more or less objc<->swift interop). It should almost be an option. I think whatever you like to do is fine by me although, I have no strong opinion on it, just some ideas.

Actually I agree with you, it definitely seems it can create different dynamics between modules. Will think about it in a future PR.